pkg/controller/clusterpool/collections.go

```diff
@@ -925,8 +925,8 @@ func (cdcs *cdcCollection) SyncClusterDeploymentCustomizationAssignments(c clien
 				}
 			}
 		} else {
-			// Ensure the finalizer is present if the CDC is not deleted, OR if it is reserved
-			if !hasFinalizer {
+			// Ensure the finalizer is present if the CDC is in this pool inventory
+			if _, ok := cdcs.byCDCName[cdc.Name]; !hasFinalizer && ok {
```

Comment:
nit: Since byCDCName contains pointers, you can get away with not using the `exists` form:
```suggestion
			if cdcs.byCDCName[cdc.Name] != nil && !hasFinalizer {
```

That said, the other collections have a `ByName()` method whose contract specifies that it returns `nil` if there's no such key (by simply wrapping the accessor as above). For parity, it would be nice to introduce that and use it (throughout).